cli/command/container/run.go

```diff
@@ -190,7 +190,11 @@ func runContainer(dockerCli *command.DockerCli, opts *runOptions, copts *contain
 		}
 
 		close, err := attachContainer(ctx, dockerCli, &errCh, config, createResponse.ID)
-		defer close()
+
+		if close != nil {
+			defer close()
+		}
```

Comment:
I think the more common way of doing this is to just move the `defer close()` after the `if err != nil`. This is only `nil` if there is an error, so handling the error first will fix the problem.